test/code/providers/runas_provider/runasprovider_test.cpp

```diff
@@ -517,10 +538,28 @@ class SCXRunAsProviderTest : public CPPUNIT_NS::TestFixture
         CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, L"run-test-unit\n", returnData.stdOut);
         CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, L"-test-\n", returnData.stdErr);
     }
+    
+    void TestDoInvokeMethodScriptTmpDir()
+        {
+            std::wstring errMsg;
+            mi::SCX_OperatingSystem_ExecuteScript_Class param;
+            param.Script_value(
+            		            "#!/bin/sh\n"
+            		            "# Get location pointed by tmpdir\n"
+            					"DIR=`dirname $0`\n"
+            		            "echo `readlink -f $DIR`\n"
```

Comment:
I don't believe readlink will work across all platforms (I don't believe it works on UNIX platforms). Did you run this in PBUILD? You must run in PBUILD to insure no build breaks.